exist-core/src/main/java/org/exist/xquery/DynamicFunctionCall.java

```diff
@@ -117,4 +128,31 @@ public void resetState(final boolean postOptimization) {
         functionExpr.resetState(postOptimization);
         arguments.forEach(arg -> arg.resetState(postOptimization));
     }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        if (functionExpr != null) {
+            builder.append(functionExpr);
+        } else {
+            builder.append("DynamicFunctionCall{arguments=");
+        }
+
+        builder.append('(');
+        if (arguments != null) {
+            for (int i = 0; i < arguments.size(); i++) {
+                if (i > 0) {
+                    builder.append(", ");
+                }
+                builder.append(arguments.get(i).toString());
+            }
```

Comment:
This could may be written in this way to reduce the amount of calls:
```
boolean firstArgument = true;
for (final Expression argument : arguments) {
    builder.append(argument.toString());
    if (firstArgument) {
      firstArgument = false;
    } else {
        builder.append(", ");
    }
}
```

Reply:
I think this will lead to one too many comma characters, i.e. a dangling comma at the end... or am I mistaken?

Reply:
Got the wrong order should be like this:
```
boolean firstArgument = true;
for (final Expression argument : arguments) {
    if (firstArgument) {
        firstArgument = false;
    } else {
        builder.append(", ");
    }
    builder.append(argument.toString());
}
```

Reply:
@reinhapa This intrigued me so I decided to benchmark it with JMH, results are here:
<img width="1535" alt="Screenshot 2022-06-10 at 17 05 05" src="https://user-images.githubusercontent.com/1264057/173095495-8d17d0af-f49c-465d-9d55-7e0231434828.png">

The JMH code looks like this: https://github.com/evolvedbinary/exist/blob/hotfix/predicate-function-reference/exist-core-jmh/src/main/java/org/exist/xquery/utils/StringJoinBenchmark.java#L48

Ultimately it seems that your and my approaches are so similar in terms of performance that the deviation indicated by JMH would account for them being identical in performance.

As the performance is the same, do you still want me to change the code?